File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/GooglepayProcessRequest.cs
using System.Text;
using CsobGatewayClientExample.Communication.DataObjects.Act;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class GooglepayProcessRequest : SignBaseRequest
{
    [JsonProperty("payId")] public string? PayId { get; set; }

    [JsonProperty("fingerprint")] public AuthData? Fingerprint { get; set; }


    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, PayId);
        Add(sb, Dttm);
        if (null != Fingerprint) Add(sb, Fingerprint.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/GooglepayProcessRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@EqualsAndHashCode(callSuper = false)
@RequiredArgsConstructor
@Data
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class GooglepayProcessRequest extends SignBase {
    @NonNull
    private String merchantId;
    @NonNull
    private String payId;
    private AuthData fingerprint;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, payId);
        add(sb, dttm);
        if (null != fingerprint) add(sb, fingerprint.toSign());
        deleteLast(sb);
        return sb.toString();
    }

}
